python/pdstools/adm/ADMDatamart.py

```diff
@@ -1654,174 +1661,325 @@ def overall_summary(
+            if name
+            else tempfile.mkdtemp(prefix="tmp_", dir=working_dir)
+        )
+        temp_dir_path = Path(temp_dir_name)
+
+        try:
+            qmd_file = "ModelReport.qmd"
+            self._copy_quarto_file(qmd_file, temp_dir_path)
+            self.save_data(temp_dir_path)
+            output_file_paths = []
+            for i, model_id in enumerate(model_list):
+                output_filename = self._get_output_filename(
+                    name, "ModelReport", model_id, output_type
                 )
-                process.communicate()
+                self._write_params_file(
+                    temp_dir_path, model_id, predictordetails_activeonly
+                )
+                self._run_quarto_command(
+                    temp_dir_path, qmd_file, output_type, output_filename, debug_mode
+                )
+                output_path = temp_dir_path / output_filename
+                if not output_path.exists():
+                    raise ValueError(f"Failed to write the report: {output_filename}")
+                output_file_paths.append(output_path)
+                if progress_callback:
+                    progress_callback(i + 1, len(model_list))
+            base_file_name = kwargs.get(
+                base_file_name, output_path
+            )  # either use the given name or the latest file name
+            file_data, file_name = cdh_utils.process_files_to_bytes(
+                output_file_paths, base_file_name
+            )
+            output_path = working_dir / file_name
+            with open(output_path, "wb") as f:
+                f.write(file_data)
+            if not output_path.exists():
+                raise ValueError(f"Failed to generate report: {output_filename}")
+
+            return output_path
+
+        except Exception as e:
+            logger.error(f"Error generating report: {str(e)}", exc_info=True)
+            raise
+        finally:
+            if not debug_mode:
+                if temp_dir_path.exists() and temp_dir_path.is_dir():
+                    shutil.rmtree(temp_dir_path, ignore_errors=True)
+
+    def generate_health_check(
+        self,
+        name: Optional[str] = None,
+        working_dir: Optional[Path] = None,
+        *,
+        output_type: str = "html",
+        **kwargs,
+    ) -> Path:
+        """
+        Generates a report based on the provided parameters.
 
-        # Main function logic
-        healthcheck_file, report = get_report_files(modelid)
-        verbose = kwargs.get("verbose", self.verbose)
+        Parameters
+        ----------
+        name : str, optional
+            The name of the report.
+        working_dir : Path, optional
+            The working directory for the output. If None, uses current working directory.
+        output_type : str, default='html'
+            The type of the output file (e.g., "html", "pdf").
+        debug_mode : bool, optional
```

Comment:
Debug mode doesn't sound like the proper argument name for this description, wouldn't it be something like `delete_temp_dir`? debug mode sounds like you'd be logging more information while running

Reply:
We already have the verbose argument for logging
1. Use verbose for both logging and keeping temp_dir
2. Create a new argument, debug_mode which overwrites verbose
`debug_mode = kwargs.get("debug_mode", self.verbose)`
So when it is set to True, it keeps logs and doesn't delete temp_dir

Reply:
Thanks for the notes @StijnKas, now all your notes are addressed except this. Do you think this is fine or should we separate the verbose and delete temp_dir logic?

Reply:
Logically I don't think they're the same thing but I follow your logic and I think it's fine to make it easier on the user. Let's stick with this.